src/components/CoreIcon/CoreIcon.spec.js

```diff
@@ -1,38 +1,39 @@
-import { shallowMount } from '@vue/test-utils'
+import { mount } from '@vue/test-utils'
 import CoreIcon from './CoreIcon.vue'
 
-/* global describe it expect */
+/* global describe it expect afterEach beforeEach */
 describe('CoreIcon.vue', () => {
-  const propsWithSizes = {
-    src: 'test-icon.png',
-    width: '32px',
-    height: '32px',
-  }
-  const propsWithoutSizes = {
-    src: 'test-icon.png',
-  }
-  it('renders correctly', () => {
-    const wrapper = shallowMount(CoreIcon, {
-      props: propsWithoutSizes,
+  let wrapper
+
+  beforeEach(() => {
+    wrapper = mount(CoreIcon, {
```

Comment:
Why you decided to use mount? In most cases, it is recommended to use shallowMount for unit tests, as it simplifies the tests, reduces the impact of external factors (no dependencies on child components) and makes them run faster.

Reply:
```
Like [mount](https://v1.test-utils.vuejs.org/api/#mount), it creates a [Wrapper](https://v1.test-utils.vuejs.org/api/wrapper) that contains the mounted and rendered Vue component, but with stubbed child components.

Notice that using shallowMount will make the component under testing different from the component you run in your application – some of its parts won't be rendered! This is why it is not the suggested way of testing components unless you face performance issues or need to simplify test arrangements.
```


This is from documentation of vue test utils, which doesn't suggest using shallowMount

basically I realized the component doesn't have children and no need for stubbing child components, so I changed it to mount

Reply:
I understand the concept, I am just wondering why we are following that strategy in developing ui components paritcularly for this project.